### DISCUSSION TITLE
ERROR  Exiting due to prerender errors.

### DISCUSSION THREAD

**Discussion by dbzx10299** [nuxt/nuxt#22397] [General]
I am using nuxt version 3.6.1, and node version 18. I have a site with nuxt content and I have looked everywhere and i see there is an error when I run `npm run generate`. I get a list of all the routes and there is one route that is causing an error. When I fetch my blog posts, the response is exactly what I would expect. All of the posts have the proper format.

If I delete the closures article, then a different article is the issue. I did set `payloadExtraction` to false and the error persists. 

<img width="612" alt="Screen Shot 2023-07-30 at 01 20 58" src="https://github.com/nuxt/nuxt/assets/40828283/d1820934-ee54-46c9-8224-4ff8480da79d">

The weird thing is that when I run the `npm run generate` command, I get one error
```js
Errors prerendering:
  ├─ /blog/posts/blog/posts/javascript-closures (500) 
//  ^^^^^^^^^^^^^^^
// why is this listed here twice?
```

This is causing the `npm run generate` command to crash and I don't understand why this is happening at all.

The stack trace looks like this
```
 ERROR  Exiting due to prerender errors.                                                                                                                         1:24:02 AM

  at prerender (node_modules/nitropack/dist/shared/nitro.1db3349c.mjs:189:11)
  at async node_modules/nuxt/dist/index.mjs:2641:7
  at async build (node_modules/nuxt/dist/index.mjs:3794:5)
  at async Object.invoke (node_modules/nuxi/dist/chunks/build.mjs:59:5)
  at async Object.invoke (node_modules/nuxi/dist/chunks/generate.mjs:36:5)
  at async _main (node_modules/nuxi/dist/cli.mjs:49:20)
```

When I look at the routes that are created I see this, the second one being highlighted in red
```
├─ /blog/posts/javascript-closures/_payload.json (2ms)                                                                                                   nitro 1:24:02 AM
├─ /blog/posts/blog/posts/javascript-closures (54ms) (Error: [500] )  
```
Any help would be appreciated, I really am stuck here not understanding why all the routes are working except this one.

**Comment by EngineerMonkey404** [comment]
Yo! Have same problem

**Reply by dbzx10299** [reply]
Take a careful look at your blog posts and check if you have a link that is wrong. This was the culprit and it confused me because of how it was showing the blog post path but the problem was simply that I had a link that was misspelled

**Comment by ejfox** [comment]
This SA answer helped me solve this problem: https://stackoverflow.com/a/76761528

**Comment by tddawson** [comment]
I'm having a very similar problem. It happened right after adding my first two blog entries. One in particular is as simple as can be but still reports:
```
Errors prerendering:
  ├─ /api/_content/query/X08pX2dCkF.1705384210024.json (17ms)                                                  nitro 9:50:31 PM  
  │ ├── Error: [404] Document not found!
  │ └── Linked from /welcome-to-my-blog
```

The Stack Overflow suggestion to turn off failOnError did let it build on Netlify again, but I'd rather get to the bottom of what's happening than suppressing the error.

**Reply by dbzx10299** [reply]
Take a close look at your links, if you have a link pointing to another article thats wrong it will fail

it looks like you have a bad link from the welcome to my blog page

**Comment by codeflorist** [comment]
It is really very very bad, that Nuxt does not provide more info as `[500]` on prerender errors. Does Nitro not have more infomation available to be able to output it?

**Reply by nathanielwarner** [reply]
Yeah this is kind of baffling. How am I supposed to debug it when the space of possible problems is just "anything that could cause it to throw a 500 during static build". Even worse is when the page in question works fine on the dev server.

**Reply by chaderenyore** [reply]
Hi, do you have a proper way of looking at this error? I still have a prerender error but no proper context.

**Reply by codeflorist** [reply]
Try it in dev mode. If it doesn't throw any error there, you have my sympathies. 🫤

**Reply by Joery-M** [reply]
For me changing `{ prerender: true }` to `{ ssr: true }` gave me more info when visiting the page.

You could also change this line in `node_modules/nitropack/dist/runtime/internal/utils.mjs:53`

```diff
  });
  const statusCode = error.statusCode || 500;
  const statusMessage = error.statusMessage ?? (statusCode === 404 ? "Not Found" : "");
  const message = !isDev && error.unhandled ? "internal server error" : error.message || error.toString();
  return {
    stack,
    statusCode,
-   statusMessage,
+   statusMessage: String(error),
    message
  };
}
```
Which gave me more info:
```
  ├─ / (1080ms)
  │ └── [500] Error: vueDemi2.effectScope is not a function
```

**Reply by alexmanzo-fundrise** [reply]
> For me changing `{ prerender: true }` to `{ ssr: true }` gave me more info when visiting the page.
> 
> You could also change this line in `node_modules/nitropack/dist/runtime/internal/utils.mjs:53`
> 
> ```diff
>   });
>   const statusCode = error.statusCode || 500;
>   const statusMessage = error.statusMessage ?? (statusCode === 404 ? "Not Found" : "");
>   const message = !isDev && error.unhandled ? "internal server error" : error.message || error.toString();
>   return {
>     stack,
>     statusCode,
> -   statusMessage,
> +   statusMessage: String(error),
>     message
>   };
> }
> ```
> 
> Which gave me more info:
> 
> ```
>   ├─ / (1080ms)
>   │ └── [500] Error: vueDemi2.effectScope is not a function
> ```

Omg THANK YOU.

**Comment by Joery-M** [comment]
I managed to debug this issue by changing the route from `{ prerender: true }` to `{ ssr: true }`, which gave more information when trying to render the page.

In my case the error was caused by an [issue in `@nuxt/pinia` cause by a version mismatch](https://github.com/vuejs/pinia/issues/2826):
```
vueDemi2.effectScope is not a function
```

I also believe the root of the empty error message is caused by [this line of code](https://github.com/nuxt/nuxt/blob/27c78381c6af44a4bfd450de8ead9ad47a376272/packages/nuxt/src/core/runtime/nitro/error.ts#L144). I'll try to reproduce the error to see what I can figure out, but no promises

**Comment by codeflorist** [comment]
It seems we are getting a fix for this at last with nitro v3 / nuxt v5:
https://github.com/nitrojs/nitro/pull/3348

Until then many poor dev-heads will continue being slammed into desks out of frustration.
